tests/test_pyxis.py

```diff
@@ -0,0 +1,426 @@
+                        "published": True
+                    }
+                ]
+            },
+            {
+                "brew": {
+                    "build": "s2i-1-2",
+                    "completion_date": "2020-08-12T11:31:39+00:00",
+                    "nvra": "s2i-1-2.arm64",
+                    "package": "s2i-core-container"
+                },
+                "repositories": [
+                    {
+                        "manifest_list_digest": "sha256:4444",
+                        "published": True
+                    }
+                ]
+            }
+        ]
+
+    def tearDown(self):
+        super().tearDown()
+        self.patcher.unpatch_all()
+        self.koji_read_config_patcher.stop()
+
+    @staticmethod
+    def copy_call_args(mock):
+        """
+        Copy args of Mock to another Mock so we can check call args if we call
+        mock with mutable args and change it between calls
+        """
+        new_mock = Mock()
+
+        def side_effect(*args, **kwargs):
+            args = deepcopy(args)
+            kwargs = deepcopy(kwargs)
+            return new_mock(*args, **kwargs)
+        mock.side_effect = side_effect
+        return new_mock
+
+    @patch('freshmaker.pyxis.HTTPKerberosAuth')
+    @patch('freshmaker.pyxis.requests.get')
+    def test_make_request(self, get, auth):
+        get.return_value = self.response
+        test_params = {'key1': 'val1'}
+        self.px._make_request('test', test_params)
+
+        get_url = self.fake_server_url + 'v1/test'
+        self.response.json.assert_called_once()
+        test_params['page_size'] = "100"
+        get.assert_called_once_with(get_url, params=test_params, auth=auth(),
+                                    timeout=conf.net_timeout)
+
+    @patch('freshmaker.pyxis.HTTPKerberosAuth')
+    @patch('freshmaker.pyxis.requests.get')
+    def test_pagination(self, get, auth):
+        my_get = self.copy_call_args(get)
+        self.response.json.side_effect = [
+            {"page": 0, "data": "fake_data1"},
+            {"page": 1, "data": "fake_data2"},
+            {"page": 2}
+        ]
+        my_get.return_value = self.response
+        test_params = {'include': ['total', 'field1']}
+        entity = 'test'
+        auth.return_value = 1
+        self.px._pagination(entity, test_params)
+
+        url = self.fake_server_url + 'v1/test'
+        self.assertEqual(get.call_count, 3)
+        default_params = {'page_size': '100', 'include': ['total', 'field1']}
+        calls = [call(url, params={**default_params, 'page': 0}, auth=1,
+                      timeout=conf.net_timeout),
+                 call(url, params={**default_params, 'page': 1}, auth=1,
+                      timeout=conf.net_timeout),
+                 call(url, params={**default_params, 'page': 2}, auth=1,
+                      timeout=conf.net_timeout)]
+        my_get.assert_has_calls(calls, any_order=True)
```

Comment:
It'd be strange if the order was incorrect. That should be a test failure.
```suggestion
        my_get.assert_has_calls(calls)
```

Reply:
I thing the problem is that if you have more calls of that method and I am testing only some of them, which are not sequential(there are calls between them), this assertion gives an error. I tried it already, before it was without 'any_order'.

Reply:
But since I will replace it with the mock of '_make_request' It's should be fine, I will remove it.